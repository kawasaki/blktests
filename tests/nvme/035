#!/bin/bash
# SPDX-License-Identifier: GPL-3.0+
# Copyright (C) 2019 Logan Gunthorpe
# Copyright (C) 2019 Eideticom Communications Inc.

. tests/nvme/rc
. common/xfs

DESCRIPTION="run mkfs and data verification fio job on an NVMeOF passthru controller"
TIMED=1

requires() {
	_nvme_requires
	_have_kernel_option NVME_TARGET_PASSTHRU
	_have_xfs
	_have_fio
}

device_requires() {
	_require_test_dev_is_not_nvme_multipath
	_require_test_dev_size "${NVME_IMG_SIZE}"
	_test_dev_suits_xfs
}

set_conditions() {
	_set_nvme_trtype "$@"
}

test_device() {
	echo "Running ${TEST_NAME}"

	_setup_nvmet

	local nsdev

	_nvmet_passthru_target_setup
	nsdev=$(_nvmet_passthru_target_connect)
	if [[ -z "$nsdev" ]]; then
		echo "FAIL: Failed to find passthru target namespace"
	elif ! _xfs_run_fio_verify_io "${nsdev}" "${NVME_IMG_SIZE}"; then
		echo "FAIL: fio verify failed"
	fi

	_nvme_disconnect_subsys
	_nvmet_passthru_target_cleanup

	echo "Test complete"
}
