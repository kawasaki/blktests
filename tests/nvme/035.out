Running nvme/035
disconnected 1 controller(s)
Test complete
